1.Introducing

Nama : Praduta Firman

Data Science : Batch 01

2.Judul/Penanda Soal
Promblems :
## Saya adalah seorang data scientist yang bekerja di salah satu perusahaan manufaktur baterai Lihtium ion. Tim produksi meminta saya untuk mengecek apakah baterai yang mereka buat dalam kondisi bagus atau tidak. Sebelum melakukan pengecekan lebih lanjut, kamu diminta menghitung hambatan/resistansi  dan daya setiap detik :
R = V / I

P = V*I

R adalah kapasitas dalam satuan ohm
P adalah daya dalam watt
I adalah kuat arus dalam satuan ampere
V adalah tegangan dalam satuan volt

Hitung hambatan baterai setiap waktunya untuk cycle ganjil dan daya baterai tiap waktunya untuk cycle genap dari dataset dengan ketentuan sebagai berikut :

(1) Gunakan function untuk menghitung hambatan dan daya

(2) Harus menerapkan konsep looping, conditional if, dan pandas query


3.Import Library :

In [1]:
import numpy as np
import pandas as pd


4.Data Loading



In [25]:
df = pd.read_excel('NMC1-9.xlsx')

In [28]:
df.head(20)

time(s)    V_m  Current (A)  Cycle       Cap
0         0  4.098        0.997      1  2.840817
1         1  4.096        0.998      1  2.840817
2         2  4.094        0.999      1  2.840817
3         3  4.093        0.998      1  2.840817
4         4  4.092        0.999      1  2.840817
5         5  4.091        0.997      1  2.840817
6         6  4.090        0.996      1  2.840817
7         7  4.089        0.996      1  2.840817
8         8  4.089        0.995      1  2.840817
9         9  4.088        0.995      1  2.840817
10       10  4.088        0.996      1  2.840817
11       11  4.087        0.996      1  2.840817
12       12  4.086        0.997      1  2.840817
13       13  4.086        0.998      1  2.840817
14       14  4.086        0.999      1  2.840817
15       15  4.085        1.000      1  2.840817
16       16  4.085        1.001      1  2.840817
17       17  4.084        1.001      1  2.840817
18       18  4.084        1.001      1  2.840817
19       19  4.083        1.000      1  2.840817

In [30]:
df['time(s)']

0            0
1            1
2            2
3            3
4            4
         ...  
90977    90977
90978    90978
90979    90979
90980    90980
90981    90981
Name: time(s), Length: 90982, dtype: int64

In [31]:
df['time(s)'].isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
90977    False
90978    False
90979    False
90980    False
90981    False
Name: time(s), Length: 90982, dtype: bool

In [32]:
df['V_m']

0        4.098
1        4.096
2        4.094
3        4.093
4        4.092
         ...  
90977    2.506
90978    2.504
90979    2.502
90980    2.500
90981    2.497
Name: V_m, Length: 90982, dtype: float64

In [34]:
df['V_m'].isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
90977    False
90978    False
90979    False
90980    False
90981    False
Name: V_m, Length: 90982, dtype: bool

In [35]:
df['Current (A)']

0        0.997
1        0.998
2        0.999
3        0.998
4        0.999
         ...  
90977    1.002
90978    1.002
90979    1.002
90980    1.002
90981    1.002
Name: Current (A), Length: 90982, dtype: float64

In [36]:
df['Current (A)'].isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
90977    False
90978    False
90979    False
90980    False
90981    False
Name: Current (A), Length: 90982, dtype: bool

In [37]:
df.isnull().sum()

time(s)        0
V_m            0
Current (A)    0
Cycle          0
Cap            0
dtype: int64

In [38]:
df.isnull().values.any()

False

In [39]:
df['Cap'].value_counts()

2.840817    10237
2.833465    10206
2.823520    10175
2.820395    10158
2.804878    10107
2.790726    10050
2.786860    10041
2.783342    10028
2.770180     9980
Name: Cap, dtype: int64

5.Answer

1.Menghitung Hambatan baterai untuk cycle ganjil

R = V/I

2.Menghitung Daya baterai untuk cycle genap

P = V*I

In [42]:
V = float(input("masukkan tegangan listrik :"))
I = float(input("masukkan besarnya arus listrik :"))

R = V/I
print("Hambatan Listrik dengan rumus R=V/I",R)


masukkan tegangan listrik :4.096
masukkan besarnya arus listrik :0.998
Hambatan Listrik dengan rumus R=V/I 4.104208416833667


In [78]:
ganjil = [1,3,5,7,9,11,13,15,17,19]
genap = [2,4,6,8,10,12,14,16,18,20]

for cycle in ganjil:
  print(cycle)

if cycle == 17:
  print(cycle)

1
3
5
7
9
11
13
15
17
19


In [64]:
V = float(input("masukkan tegangan listrik :"))
I = float(input("masukkan arus listrik :"))

P = V*I
print("Daya listrik dengan rumus P=V*I",P)

masukkan tegangan listrik :4.094
masukkan arus listrik :0.999
Daya listrik dengan rumus P=V*I 4.089906


In [79]:
genap = [2,4,6,8,10,12,14,16,18,20]

for cycle in genap:
  print(cycle)

2
4
6
8
10
12
14
16
18
20


Pandas Query

In [81]:
df

time(s)    V_m  Current (A)  Cycle       Cap
0            0  4.098        0.997      1  2.840817
1            1  4.096        0.998      1  2.840817
2            2  4.094        0.999      1  2.840817
3            3  4.093        0.998      1  2.840817
4            4  4.092        0.999      1  2.840817
...        ...    ...          ...    ...       ...
90977    90977  2.506        1.002      9  2.770180
90978    90978  2.504        1.002      9  2.770180
90979    90979  2.502        1.002      9  2.770180
90980    90980  2.500        1.002      9  2.770180
90981    90981  2.497        1.002      9  2.770180

[90982 rows x 5 columns]

In [82]:
df.tail()

time(s)    V_m  Current (A)  Cycle      Cap
90977    90977  2.506        1.002      9  2.77018
90978    90978  2.504        1.002      9  2.77018
90979    90979  2.502        1.002      9  2.77018
90980    90980  2.500        1.002      9  2.77018
90981    90981  2.497        1.002      9  2.77018

6.Hasil

R = V/I
pada cycle ganjil di antaranya

R = 4.096/0.998
  = 4.10420

P = V*I
pada cycle genap di antaranya

P = 4.094 * 0.999
  = 4.0899